auslib/web/api_validator.py

```diff
@@ -0,0 +1,30 @@
+import codecs
+import logging
+from connexion.decorators.validation import ParameterValidator
+
+log = logging.getLogger(__name__)
+
+
+class BalrogParameterValidator(ParameterValidator):
+    """
+    This class is used to keep the flask "defaults" feature behavior on spec validation level.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(BalrogParameterValidator, self).__init__(*args, **kwargs)
+
+    def validate_path_parameter(self, param, request):
+        # Keep the base class behavior.
+        val = request.path_params.get(param['name'].replace('-', '_'))
+
+        # Supporting the flask add_url_rule "defaults" parameter to the swagger path parameter.
+        if val is None and 'default' in param:
+            val = param['default']
+            request.path_params[param['name']] = val
+
+        # Encode strings in utf-8.
+        if param['type'] == 'string':
+            val = codecs.encode(val, 'utf-8')
```

Comment:
I had considered these two resoures before using utf-8:

-https://docs.python.org/2/howto/unicode.html#encodings
"UTF-8 has several convenient properties:
1) It can handle any Unicode code point."

and

-http://flask.pocoo.org/docs/0.12/unicode/

Today I took a look in werkzeug docs and found this:
https://github.com/pallets/werkzeug/blob/0bc61df6e1ae9f2ffdf5d066aa3cd9d5ebcb307d/docs/unicode.rst#unicode-in-http

Seems that there is a open PR to handle strings as UTF-8 too in connexion:
zalando/connexion#349